### ISSUE TITLE
Exception when using Tooltip demo in Flutter Gallery

### ISSUE THREAD

**Issue by abarth** [flutter/flutter#4850]
```
══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
The following assertion was thrown during _updateSemantics():
'package:flutter/src/rendering/semantics.dart': Failed assertion: line 365 pos 14:
'!owner._dirtyNodes.contains(this)' is not true.
When the exception was thrown, this was the stack:
#0      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:27)
#1      _AssertionError._checkAssertion (dart:core-patch/errors_patch.dart:34)
#2      SemanticsNode._markDirty (package:flutter/src/rendering/semantics.dart:365:14)
#3      SemanticsNode.attach (package:flutter/src/rendering/semantics.dart:336:7)
#4      SemanticsNode.attach (package:flutter/src/rendering/semantics.dart:342:15)
#5      AbstractNode.adoptChild (package:flutter/src/rendering/node.dart:115:13)
#6      SemanticsNode.finalizeChildren (package:flutter/src/rendering/semantics.dart:288:11)
#7      _InterestingSemanticsFragment.compile.<compile_sync_body> (package:flutter/src/rendering/object.dart:596:12)
#8      _SyncIterator.moveNext (dart:core-patch/core_patch.dart:65)
#9      Iterable.single (dart:core/iterable.dart:494)
#10     RenderObject._updateSemantics (package:flutter/src/rendering/object.dart:2008:82)
#11     PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:931:16)
#12     BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding.beginFrame (package:flutter/src/rendering/binding.dart:161:19)
#13     BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding&WidgetsBinding.beginFrame (package:flutter/src/widgets/binding.dart:210:21)
#14     BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:149:5)
#15     BindingBase&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:476:15)
#16     BindingBase&SchedulerBinding.handleBeginFrame (package:flutter/src/scheduler/binding.dart:436:7)
#17     _beginFrame (file:///b/build/slave/Mac_Engine/build/src/out/ios_debug_sim/gen/sky/bindings/dart_ui/hooks.dart:48)
The following RenderObject was being processed when the exception was fired:
  RenderSemanticAnnotations
This RenderObject had the following creator information:
  Semantics ← Focus-[GlobalObjectKey MaterialPageRoute<Null>(927337741)] ← _ModalScope-[GlobalKey
  642064156] ← _OverlayEntry-[GlobalKey 447812396] ← Stack ← Overlay-[GlobalKey 129519427] ←
  _FocusScope ← Semantics ← Focus-[GlobalKey 741535956] ← AbsorbPointer ← ⋯
This RenderObject had the following descendants (showing up to depth 5):
  RenderRepaintBoundary
    RenderIgnorePointer
      RenderTransform
        RenderOpacity
          RenderDecoratedBox
════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by abarth** [comment]
When trying to reproduce https://github.com/flutter/flutter/issues/4640

**Comment by abarth** [comment]
Actually, just when openning the "components" menu on the home screen.
